pyside cmake code does not handle crossbuild on pyi files generation (type checker stub file).
So disable their generation on crossbuild.

--- a/sources/pyside6/cmake/Macros/PySideModules.cmake.ORIG	2021-05-05 08:27:28.000000000 +0200
+++ b/sources/pyside6/cmake/Macros/PySideModules.cmake	2021-05-20 11:35:43.693098101 +0200
@@ -204,18 +204,24 @@
     endif()
 
     # Add target to generate pyi file, which depends on the module target.
+    if(NOT CMAKE_CROSSCOMPILING)
     add_custom_target("${module_NAME}_pyi" ALL
                       COMMAND ${CMAKE_COMMAND} -E env ${ld_prefix}
                       "${SHIBOKEN_PYTHON_INTERPRETER}"
                       "${CMAKE_CURRENT_SOURCE_DIR}/../support/generate_pyi.py" ${generate_pyi_options})
     add_dependencies("${module_NAME}_pyi" ${module_NAME})
+    else()
+        message(STATUS "crossbuild: disable generation of ${module_NAME}_pyi")
+    endif()
 
     # install
     install(TARGETS ${module_NAME} LIBRARY DESTINATION "${PYTHON_SITE_PACKAGES}/PySide6")
 
+    if(NOT CMAKE_CROSSCOMPILING)
     file(GLOB hinting_stub_files RELATIVE "${CMAKE_CURRENT_BINARY_DIR}/PySide6" "${CMAKE_CURRENT_BINARY_DIR}/PySide6/*.pyi")
     install(FILES ${hinting_stub_files}
             DESTINATION "${PYTHON_SITE_PACKAGES}/PySide6")
+    endif()
 
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PySide6/${module_NAME}/pyside6_${lower_module_name}_python.h
             DESTINATION include/PySide6${pyside6_SUFFIX}/${module_NAME}/)
